packages/program/src/functions/multicall-scope.ts

```diff
@@ -10,11 +10,11 @@ export class MultiCallInvocationScope<TReturn = any> extends BaseInvocationScope
     this.addCalls(funcScopes);
```

Comment:
This is a problem. The `constructor` of `MultiCallInvocationScope` calls `addCalls`, which is now an `async` function. This might cause issues. Not sure how to go around this.

I have tried several different approaches including the one shared by @LuizAsFight in #808, but even in that approach eventually I ended up in a place where I had to pass in `gasPriceFactor` to the constructor of `InvocationScope`s. This was also not possible because these constructors were being called from other constructors.